app/models/organization.rb

```diff
@@ -127,5 +127,27 @@ def central
     def central_url
       ApplicationRoot.subdominated('central').to_s
     end
+
+    def create_from_json!(json)
+      organization_json = parse_json json
+      Organization.create! organization_json
+    end
+
+    def update_from_json!(json)
+      organization_json = parse_json json
+
+      organization = Organization.find_by! name: organization_json[:name]
+      organization.update_attributes organization_json
```

Comment:
Use just `update` - or better `update!`, which is shorthand for `update_attributes`. 

Also, `update` actually persist the object, so there is no need of using `save!`. 

To sum up, just use `organization.update! organization_json`, and no `save!`